src/openzaak/components/catalogi/admin/admin_views.py

```diff
@@ -88,8 +104,21 @@ def get_context_data(self, **kwargs):
 
         iotypen = self.request.session.get("iotypen")
         if iotypen:
+            iot_dict = self.get_query_dict(
+                InformatieObjectType.objects.filter(catalogus=catalogus)
+            )
+            iotypen = sorted(iotypen, key=lambda x: x["omschrijving"])
             iotype_forms = InformatieObjectTypeFormSet(
-                initial=[{"new_instance": instance} for instance in iotypen],
+                initial=[
+                    {
+                        "existing": self.find_existing(
+                            iot_dict,
+                            instance["omschrijving"],
+                            instance["begin_geldigheid"],
```

Comment:
@annashamray I'm not sure I inderstand the difference between your suggestion and what Conor did. We don't want to match on earlier versions of the informatieobjecttype, how this is done is up to you and Conor.

Reply:
This is how I understand. Please correct me if I'm wrong
Example: 
We want to import iotype1 with `omschrijving` = "invoice", `begin_geldigheid` = "01-01-2023" and `einde_geldigheid` = None. In the target catalogus there is already iotype2 with the same `omschrijving` = "invoice", but with `begin_geldigheid` = "02-01-2023" and `einde_geldigheid` = None. 
In the current implementation iotype2 won't be preselected. But I think it should

I suggest to prefilter queryset, so it will have objects with unique `omschrijving` and last `begin_geldigheid`, therefore we will make sure to preselect only the latest versions of iotypen

Reply:
I agree with you Anna, a different 'begin_geldigheid' shouldn't cause the iotype to not be preselected.

Reply:
What if you are importing multiple IOTs? Then it would be the same for both of them. 

Also IOTs do not display their start time so anything with the same `omschrijving` would have the exact same label.

Reply:
1. I want to assume that the clients won't want to import outdated iotypen, but even if they do let's not encourage it and make them select old iotypen manually

2. I think that showing `begin_geldigheid` in the label in selector options is a good idea

@alextreme what do you think?

Reply:
I'm not sure how relevant this discussion is, as far as I can see nearly all iot's are published without an enddate (aside from changing the vertrouwelijkheidsaanduiding there isn't a reason to update an iot).

Auto-select the most recent matching iot and show begin_geldigheid in the label selector options.

Reply:
Is there a specific format wanted? Currently I've put it as `Adreslijst 2023-01-01 (CONCEPT)`

Reply:
That works for me